formats/html.ts

```diff
@@ -1,44 +1,358 @@
+
+		this.showResult(result);
+		if (result.errors.length > 0) {
+			console.error(result.errors);
+		}
+	}
+
+	async processFile(result: ImportResult & { errors: unknown[] }, folder: TFolder, file: PickedFile) {
+		let mdFile: TFile | null = null;
+		try {
+			++result.total;
+			const htmlContent = await file.readText();
+			mdFile = await this.saveAsMarkdownFile(folder, file.basename, "");
+
+			const dom = new DOMParser().parseFromString(htmlContent, "text/html");
+			const pathURL = file instanceof NodePickedFile ? nodeUrl.pathToFileURL(file.filepath) : undefined;
+			const downloads = await Promise.allSettled(
+				Array.from(dom.querySelectorAll<HTMLAudioElement | HTMLImageElement | HTMLVideoElement>("audio, img, video"))
+					.map(async element => {
+						type TagNames = keyof {
+							[K in keyof HTMLElementTagNameMap as HTMLElementTagNameMap[K] extends typeof element ? K : never]: never
+						};
+						let src = "";
+						try {
+							src = element.getAttribute("src"); // `element.src` does not give the raw `src` string
+							return [
+								src,
+								await this.downloadAttachmentCached(
+									mdFile,
+									element.tagName.toLowerCase() as TagNames,
+									new URL(src.startsWith("//") ? `https:${src}` : src, pathURL),
+								),
+							] as const;
+						} catch (e) {
+							result.errors.push(e);
+							throw src;
+						}
+					})
+			);
+			const links = Array.from(dom.querySelectorAll("a")).map(a => a.getAttribute("href")).filter(href => href);
+			result.total += downloads.length;
+			result.failed = result.failed.concat(downloads
+				.filter((dl): dl is typeof dl & { status: "rejected" } => dl.status === "rejected")
+				.map(({ reason }) => decodeURIComponent(reason)));
+			result.skipped = result.skipped.concat(downloads
+				.filter((dl): dl is typeof dl & { status: "fulfilled" } => dl.status === "fulfilled" && !dl.value[1])
+				.map(({ value: [src] }) => decodeURIComponent(src)));
+
+			let mdContent = htmlToMarkdown(htmlContent);
+			if (mdContent) {
+				const attachments = Object.fromEntries(downloads
+					.filter((dl): dl is typeof dl & { status: "fulfilled" } => dl.status === "fulfilled" && Boolean(dl.value[1]))
+					.map(({ value }) => value));
+				if (Object.keys(attachments).length > 0) {
+					mdContent = mdContent.replace(
+						alternativeRegExp([...links, ...Object.keys(attachments)]),
+						sstr => sstr.replace(/ /gu, "%20"),
+					);
+					const attachments2 = Object.fromEntries(Object.entries(attachments).map(([key, value]) => [decodeURIComponent(key), value]));
+
+					const cache = new Promise<CachedMetadata>(resolve => {
+						const ref = this.app.metadataCache.on("changed", (file, _1, cache) => {
+							if (file.path === mdFile.path) {
+								try {
+									resolve(cache);
+								} finally {
+									this.app.metadataCache.offref(ref);
+								}
+							}
+						});
+					});
+					await this.app.vault.process(mdFile, data => `${data}${mdContent}`);
```

Comment:
Since we saved the file as empty before, we should just call modify here instead of process.

Reply:
Done. I was thinking about plugins that write text to newly created files.